// ant-design-x-vue/es/actions/style/index.mjs
import "vue";
import "../../_util/cssinjs/StyleContext.mjs";
import "../../_util/cssinjs/theme/ThemeCache.mjs";
import "../../_util/warning.mjs";
import "../../_util/cssinjs/transformers/legacyLogicalProperties.mjs";
import { merge as e } from "../../_util/cssinjs-utils/util/statistic.mjs";
import { genStyleHooks as t } from "../../theme/genStyleUtils.mjs";
const n = (i) => {
  const { componentCls: o, calc: r } = i;
  return {
    [o]: {
      [`&${o}-rtl`]: {
        direction: "rtl"
      },
      [`${o}-list`]: {
        display: "inline-flex",
        flexDirection: "row",
        gap: i.paddingXS,
        color: i.colorTextDescription,
        "&-item, &-sub-item": {
          cursor: "pointer",
          padding: i.paddingXXS,
          borderRadius: i.borderRadius,
          height: i.controlHeightSM,
          width: i.controlHeightSM,
          boxSizing: "border-box",
          display: "inline-flex",
          alignItems: "center",
          justifyContent: "center",
          "&-icon": {
            display: "inline-flex",
            alignItems: "center",
            justifyContent: "center",
            fontSize: i.fontSize,
            width: "100%",
            height: "100%"
          },
          "&:hover": {
            background: i.colorBgTextHover
          }
        }
      },
      "& .border": {
        padding: `${i.paddingXS}px ${i.paddingSM}px`,
        gap: i.paddingSM,
        borderRadius: r(i.borderRadiusLG).mul(1.5).equal(),
        backgroundColor: i.colorBorderSecondary,
        color: i.colorTextSecondary,
        [`${o}-list-item, ${o}-list-sub-item`]: {
          padding: 0,
          lineHeight: i.lineHeight,
          "&-icon": {
            fontSize: i.fontSizeLG
          },
          "&:hover": {
            opacity: 0.8
          }
        }
      },
      "& .block": {
        display: "flex"
      }
    }
  };
}, d = () => ({}), u = t(
  "Actions",
  (i) => {
    const o = e(i, {});
    return [n(o)];
  },
  d
);
export {
  u as default,
  d as prepareComponentToken
};
