// ant-design-x-vue/es/sender/StopLoading.mjs
import { defineComponent as n, createVNode as e, createTextVNode as s } from "vue";
import { _ as i } from "../chunks/helper-chunk.mjs";
const a = /* @__PURE__ */ n({
  name: "AXSenderStopLoading",
  __name: "StopLoading",
  props: {
    className: {
      type: String,
      required: !0
    }
  },
  setup(r, {
    expose: t
  }) {
    t();
    const o = r;
    return () => e("svg", {
      color: "currentColor",
      viewBox: "0 0 1000 1000",
      xmlns: "http://www.w3.org/2000/svg",
      xmlnsXlink: "http://www.w3.org/1999/xlink",
      class: o.className
    }, [e("title", null, [s("Stop Loading")]), e("rect", {
      fill: "currentColor",
      height: "250",
      rx: "24",
      ry: "24",
      width: "250",
      x: "375",
      y: "375"
    }, null), e("circle", {
      cx: "500",
      cy: "500",
      fill: "none",
      r: "450",
      stroke: "currentColor",
      "stroke-width": "100",
      opacity: "0.45"
    }, null), e("circle", {
      cx: "500",
      cy: "500",
      fill: "none",
      r: "450",
      stroke: "currentColor",
      "stroke-width": "100",
      "stroke-dasharray": "600 9999999"
    }, [e("animateTransform", {
      attributeName: "transform",
      dur: "1s",
      from: "0 500 500",
      repeatCount: "indefinite",
      to: "360 500 500",
      type: "rotate"
    }, null)])]);
  }
}), u = /* @__PURE__ */ i(a, [["__file", "/Users/wangzhichao/办公/code/github/ant-design-x-vue-pure/ant-design-x-vue/src/sender/StopLoading.vue"]]);
export {
  u as default
};
