// ant-design-x-vue/es/welcome/style/index.mjs
import "vue";
import "../../_util/cssinjs/StyleContext.mjs";
import "../../_util/cssinjs/theme/ThemeCache.mjs";
import "../../_util/warning.mjs";
import "../../_util/cssinjs/transformers/legacyLogicalProperties.mjs";
import { merge as r } from "../../_util/cssinjs-utils/util/statistic.mjs";
import { genStyleHooks as d } from "../../theme/genStyleUtils.mjs";
const l = (e) => {
  const { componentCls: i, calc: t } = e, n = t(e.fontSizeHeading3).mul(e.lineHeightHeading3).equal(), a = t(e.fontSize).mul(e.lineHeight).equal();
  return {
    [i]: {
      gap: e.padding,
      // ======================== Icon ========================
      [`${i}-icon`]: {
        height: t(n).add(a).add(e.paddingXXS).equal(),
        display: "flex",
        img: {
          height: "100%"
        }
      },
      // ==================== Content Wrap ====================
      [`${i}-content-wrapper`]: {
        gap: e.paddingXS,
        flex: "auto",
        minWidth: 0,
        [`${i}-title-wrapper`]: {
          gap: e.paddingXS
        },
        [`${i}-title`]: {
          margin: 0
        },
        [`${i}-extra`]: {
          marginInlineStart: "auto"
        }
      }
    }
  };
}, o = (e) => {
  const { componentCls: i } = e;
  return {
    [i]: {
      // ======================== Filled ========================
      "&-filled": {
        paddingInline: e.padding,
        paddingBlock: e.paddingSM,
        background: e.colorFillContent,
        borderRadius: e.borderRadiusLG
      },
      // ====================== Borderless ======================
      "&-borderless": {
        [`${i}-title`]: {
          fontSize: e.fontSizeHeading3,
          lineHeight: e.lineHeightHeading3
        }
      }
    }
  };
}, g = () => ({}), H = d(
  "Welcome",
  (e) => {
    const i = r(e, {});
    return [l(i), o(i)];
  },
  g
);
export {
  H as default,
  g as prepareComponentToken
};
