// ant-design-x-vue/es/x-provider/index-chunk.mjs
import { defineComponent as p, useSlots as c, computed as n, createVNode as s, mergeProps as f } from "vue";
import { c as m, _ as y } from "../chunks/helper-chunk.mjs";
import { XProviderContextProvider as q } from "./context.mjs";
import g from "./hooks/use-x-provider-context.mjs";
import { ConfigProvider as h } from "ant-design-vue";
const v = /* @__PURE__ */ p({
  name: "AXProvider",
  inheritAttrs: !1,
  __name: "index",
  props: {
    actions: {
      type: null,
      required: !1
    },
    bubble: {
      type: null,
      required: !1
    },
    conversations: {
      type: null,
      required: !1
    },
    prompts: {
      type: null,
      required: !1
    },
    sender: {
      type: null,
      required: !1
    },
    suggestion: {
      type: null,
      required: !1
    },
    thoughtChain: {
      type: null,
      required: !1
    },
    attachments: {
      type: null,
      required: !1
    },
    welcome: {
      type: null,
      required: !1
    },
    iconPrefixCls: {
      type: String,
      required: !1
    },
    getTargetContainer: {
      type: Function,
      required: !1
    },
    getPopupContainer: {
      type: Function,
      required: !1
    },
    prefixCls: {
      type: String,
      required: !1
    },
    getPrefixCls: {
      type: Function,
      required: !1
    },
    renderEmpty: {
      type: null,
      required: !1
    },
    transformCellText: {
      type: Function,
      required: !1
    },
    csp: {
      type: null,
      required: !1
    },
    input: {
      type: Object,
      required: !1
    },
    autoInsertSpaceInButton: {
      type: Boolean,
      required: !1
    },
    locale: {
      type: null,
      required: !1
    },
    pageHeader: {
      type: Object,
      required: !1
    },
    componentSize: {
      type: null,
      required: !1
    },
    componentDisabled: {
      type: Boolean,
      required: !1
    },
    direction: {
      type: String,
      required: !1
    },
    space: {
      type: Object,
      required: !1
    },
    virtual: {
      type: Boolean,
      required: !1
    },
    dropdownMatchSelectWidth: {
      type: null,
      required: !1
    },
    form: {
      type: Object,
      required: !1
    },
    pagination: {
      type: Object,
      required: !1
    },
    theme: {
      type: null,
      required: !1
    },
    select: {
      type: Object,
      required: !1
    },
    wave: {
      type: Object,
      required: !1
    }
  },
  setup(e, {
    expose: o
  }) {
    o();
    const l = m(e, ["attachments", "bubble", "conversations", "prompts", "sender", "suggestion", "thoughtChain", "welcome"]), t = c(), a = n(() => ({
      attachments: e.attachments,
      bubble: e.bubble,
      conversations: e.conversations,
      prompts: e.prompts,
      sender: e.sender,
      suggestion: e.suggestion,
      thoughtChain: e.thoughtChain,
      welcome: e.welcome
    })), {
      theme: r
    } = g(), u = n(() => ({
      ...(r == null ? void 0 : r.value) || {},
      ...l.theme
    })), d = n(() => {
      var i;
      return (i = t.default) == null ? void 0 : i.call(t);
    });
    return () => s(q, {
      value: a.value
    }, {
      default: () => [s(h, f(l, {
        theme: u.value
      }), {
        default: () => [d.value]
      })]
    });
  }
}), w = /* @__PURE__ */ y(v, [["__file", "/Users/wangzhichao/办公/code/github/ant-design-x-vue-pure/ant-design-x-vue/src/x-provider/index.vue"]]);
export {
  w as X
};
