// ant-design-x-vue/es/_util/cssinjs/index.mjs
import r from "./hooks/useCacheToken.mjs";
import { default as W } from "./hooks/useCSSVarRegister.mjs";
import e, { extractStyle as o } from "./hooks/useStyleRegister/index.mjs";
import t from "./Keyframes.mjs";
import m from "./linters/legacyNotSelectorLinter.mjs";
import i from "./linters/logicalPropertiesLinter.mjs";
import a from "./linters/parentSelectorLinter.mjs";
import { StyleProvider as s, useStyleProvider as p, useStyleInject as n, createCache as f } from "./StyleContext.mjs";
import l from "./theme/createTheme.mjs";
import c from "./theme/Theme.mjs";
import "./theme/ThemeCache.mjs";
import S from "./transformers/legacyLogicalProperties.mjs";
import y from "./transformers/px2rem.mjs";
import { supportWhere as g, supportLogicProps as u } from "./util.mjs";
import { token2CSSVar as b, unit as q } from "./util.mjs";
const R = {
  Theme: c,
  createTheme: l,
  useStyleRegister: e,
  useCacheToken: r,
  createCache: f,
  useStyleInject: n,
  useStyleProvider: p,
  Keyframes: t,
  extractStyle: o,
  // Transformer
  legacyLogicalPropertiesTransformer: S,
  px2remTransformer: y,
  // Linters
  logicalPropertiesLinter: i,
  legacyNotSelectorLinter: m,
  parentSelectorLinter: a,
  // cssinjs
  StyleProvider: s
}, V = {
  supportModernCSS: () => g() && u()
};
export {
  t as Keyframes,
  s as StyleProvider,
  c as Theme,
  V as _experimental,
  f as createCache,
  l as createTheme,
  R as default,
  o as extractStyle,
  S as legacyLogicalPropertiesTransformer,
  m as legacyNotSelectorLinter,
  i as logicalPropertiesLinter,
  a as parentSelectorLinter,
  y as px2remTransformer,
  b as token2CSSVar,
  q as unit,
  W as useCSSVarRegister,
  r as useCacheToken,
  n as useStyleInject,
  p as useStyleProvider,
  e as useStyleRegister
};
