// ant-design-x-vue/es/sender/components/ClearButton.mjs
import { defineComponent as n, mergeDefaults as t, createVNode as l, mergeProps as a } from "vue";
import { c as s, _ as i } from "../../chunks/helper-chunk.mjs";
import d from "./ActionButton/index.mjs";
import { n as o } from "../../chunks/module-chunk.mjs";
const p = /* @__PURE__ */ n({
  name: "AXSenderClearButton",
  __name: "ClearButton",
  props: t({
    prefixCls: {
      type: null,
      required: !1
    },
    type: {
      type: null,
      required: !1
    },
    htmlType: {
      type: null,
      required: !1
    },
    shape: {
      type: null,
      required: !1
    },
    size: {
      type: null,
      required: !1
    },
    loading: {
      type: null,
      required: !1
    },
    disabled: {
      type: null,
      required: !1
    },
    ghost: {
      type: null,
      required: !1
    },
    block: {
      type: null,
      required: !1
    },
    danger: {
      type: null,
      required: !1
    },
    icon: {
      type: null,
      required: !1
    },
    href: {
      type: null,
      required: !1
    },
    target: {
      type: null,
      required: !1
    },
    title: {
      type: null,
      required: !1
    },
    onClick: {
      type: null,
      required: !1
    },
    onMousedown: {
      type: null,
      required: !1
    },
    audioIcon: {
      required: !1
    },
    audioDisabledIcon: {
      required: !1
    },
    audioRecordingIcon: {
      required: !1
    }
  }, {
    disabled: void 0
  }),
  setup(e, {
    expose: r
  }) {
    r();
    const u = s(e, ["disabled"]);
    return () => l(d, a({
      disabled: e.disabled
    }, u, {
      action: "onClear"
    }), {
      default: () => [l(o, null, null)]
    });
  }
}), m = /* @__PURE__ */ i(p, [["__file", "/Users/wangzhichao/办公/code/github/ant-design-x-vue-pure/ant-design-x-vue/src/sender/components/ClearButton.vue"]]);
export {
  m as default
};
