// ant-design-x-vue/es/sender/components/SendButton.mjs
import { defineComponent as t, mergeDefaults as u, createVNode as r, mergeProps as s } from "vue";
import { c as a, _ as d } from "../../chunks/helper-chunk.mjs";
import i from "./ActionButton/index.mjs";
import { A as o } from "../../chunks/module-chunk.mjs";
const p = /* @__PURE__ */ t({
  name: "AXSenderLoadingButton",
  __name: "SendButton",
  props: u({
    prefixCls: {
      type: null,
      required: !1
    },
    type: {
      type: null,
      required: !1
    },
    htmlType: {
      type: null,
      required: !1
    },
    shape: {
      type: null,
      required: !1
    },
    size: {
      type: null,
      required: !1
    },
    loading: {
      type: null,
      required: !1
    },
    disabled: {
      type: null,
      required: !1
    },
    ghost: {
      type: null,
      required: !1
    },
    block: {
      type: null,
      required: !1
    },
    danger: {
      type: null,
      required: !1
    },
    icon: {
      type: null,
      required: !1
    },
    href: {
      type: null,
      required: !1
    },
    target: {
      type: null,
      required: !1
    },
    title: {
      type: null,
      required: !1
    },
    onClick: {
      type: null,
      required: !1
    },
    onMousedown: {
      type: null,
      required: !1
    },
    audioIcon: {
      required: !1
    },
    audioDisabledIcon: {
      required: !1
    },
    audioRecordingIcon: {
      required: !1
    }
  }, {
    type: "primary",
    shape: "circle",
    disabled: void 0
  }),
  setup(e, {
    expose: l
  }) {
    l();
    const n = a(e, ["type", "shape", "disabled"]);
    return () => r(i, s({
      type: e.type,
      shape: e.shape,
      disabled: e.disabled
    }, n, {
      action: "onSend"
    }), {
      default: () => [r(o, null, null)]
    });
  }
}), m = /* @__PURE__ */ d(p, [["__file", "/Users/wangzhichao/办公/code/github/ant-design-x-vue-pure/ant-design-x-vue/src/sender/components/SendButton.vue"]]);
export {
  m as default
};
