// ant-design-x-vue/es/attachments/FileList/Progress.mjs
import { defineComponent as o, createVNode as e, createTextVNode as u } from "vue";
import { theme as a, Progress as i } from "ant-design-vue";
import { _ as n } from "../../chunks/helper-chunk.mjs";
const p = /* @__PURE__ */ o({
  name: "AXAttachmentsProgress",
  __name: "Progress",
  props: {
    prefixCls: {
      type: String,
      required: !1
    },
    type: {
      type: null,
      required: !1
    },
    percent: {
      type: Number,
      required: !1
    },
    format: {
      type: null,
      required: !1
    },
    status: {
      type: null,
      required: !1
    },
    showInfo: {
      type: Boolean,
      required: !1
    },
    strokeWidth: {
      type: Number,
      required: !1
    },
    strokeLinecap: {
      type: null,
      required: !1
    },
    strokeColor: {
      type: null,
      required: !1
    },
    trailColor: {
      type: String,
      required: !1
    },
    width: {
      type: Number,
      required: !1
    },
    success: {
      type: null,
      required: !1
    },
    gapDegree: {
      type: Number,
      required: !1
    },
    gapPosition: {
      type: null,
      required: !1
    },
    size: {
      type: null,
      required: !1
    },
    steps: {
      type: Number,
      required: !1
    },
    successPercent: {
      type: Number,
      required: !1
    },
    title: {
      type: String,
      required: !1
    },
    progressStatus: {
      type: null,
      required: !1
    }
  },
  setup(r, {
    expose: t
  }) {
    t();
    const {
      token: s
    } = a.useToken();
    return () => e(i, {
      type: "circle",
      percent: r.percent,
      size: s.value.fontSizeHeading2 * 2,
      strokeColor: "#FFF",
      trailColor: "rgba(255, 255, 255, 0.3)",
      format: (l) => e("span", {
        style: {
          color: "#FFF"
        }
      }, [(l || 0).toFixed(0), u("%")])
    }, null);
  }
}), y = /* @__PURE__ */ n(p, [["__file", "/Users/wangzhichao/办公/code/github/ant-design-x-vue-pure/ant-design-x-vue/src/attachments/FileList/Progress.vue"]]);
export {
  y as default
};
