// ant-design-x-vue/es/sender/style/header.mjs
const t = (d) => {
  const { componentCls: e, calc: o } = d, r = `${e}-header`;
  return {
    [e]: {
      [r]: {
        borderBottomWidth: d.lineWidth,
        borderBottomStyle: "solid",
        borderBottomColor: d.colorBorder,
        // ======================== Header ========================
        "&-header": {
          background: d.colorFillAlter,
          fontSize: d.fontSize,
          lineHeight: d.lineHeight,
          paddingBlock: o(d.paddingSM).sub(d.lineWidthBold).equal(),
          paddingInlineStart: d.padding,
          paddingInlineEnd: d.paddingXS,
          display: "flex",
          borderRadius: {
            _skip_check_: !0,
            value: o(d.borderRadius).mul(2).equal()
          },
          borderBottomRightRadius: 0,
          borderBottomLeftRadius: 0,
          [`${r}-title`]: {
            flex: "auto"
          }
        },
        // ======================= Content ========================
        "&-content": {
          padding: d.padding
        },
        // ======================== Motion ========================
        "&-motion": {
          transition: ["height", "border"].map((i) => `${i} ${d.motionDurationSlow}`).join(","),
          overflow: "hidden",
          "&-enter-start, &-leave-active": {
            borderBottomColor: "transparent"
          },
          "&-hidden": {
            display: "none"
          }
        }
      }
    }
  };
};
export {
  t as default
};
