// ant-design-x-vue/es/components.ts-chunk.mjs
import "./x-provider/index.mjs";
import o from "./bubble/index.mjs";
import "./conversations/index.mjs";
import "./welcome/index.mjs";
import "./prompts/index.mjs";
import "./suggestion/index.mjs";
import r from "./sender/index.mjs";
import t from "./attachments/index.mjs";
import "./thought-chain/index.mjs";
import "./actions/index.mjs";
import m, { defaultPrefixCls as i } from "./x-provider/hooks/use-x-provider-context.mjs";
import { X as e } from "./x-provider/index-chunk.mjs";
import p from "./bubble/BubbleList.mjs";
import f from "./conversations/Conversations.mjs";
import n from "./welcome/Welcome.mjs";
import s from "./prompts/Prompts.mjs";
import l from "./suggestion/Suggestion.mjs";
import u from "./thought-chain/ThoughtChain.mjs";
import a from "./actions/Actions.mjs";
const L = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  Actions: a,
  Attachments: t,
  Bubble: o,
  BubbleList: p,
  Conversations: f,
  Prompts: s,
  Sender: r,
  Suggestion: l,
  ThoughtChain: u,
  Welcome: n,
  XProvider: e,
  defaultPrefixCls: i,
  useXProviderContext: m
}, Symbol.toStringTag, { value: "Module" }));
export {
  L as c
};
