// ant-design-x-vue/es/components.mjs
import "./x-provider/index.mjs";
import { default as u } from "./bubble/index.mjs";
import "./conversations/index.mjs";
import "./welcome/index.mjs";
import "./prompts/index.mjs";
import "./suggestion/index.mjs";
import { default as d } from "./sender/index.mjs";
import { default as x } from "./attachments/index.mjs";
import "./thought-chain/index.mjs";
import "./actions/index.mjs";
import { defaultPrefixCls as b, default as h } from "./x-provider/hooks/use-x-provider-context.mjs";
import { X as P } from "./x-provider/index-chunk.mjs";
import { default as g } from "./bubble/BubbleList.mjs";
import { default as X } from "./conversations/Conversations.mjs";
import { default as B } from "./welcome/Welcome.mjs";
import { default as L } from "./prompts/Prompts.mjs";
import { default as W } from "./suggestion/Suggestion.mjs";
import { default as k } from "./thought-chain/ThoughtChain.mjs";
import { default as w } from "./actions/Actions.mjs";
export {
  w as Actions,
  x as Attachments,
  u as Bubble,
  g as BubbleList,
  X as Conversations,
  L as Prompts,
  d as Sender,
  W as Suggestion,
  k as ThoughtChain,
  B as Welcome,
  P as XProvider,
  b as defaultPrefixCls,
  h as useXProviderContext
};
