// ant-design-x-vue/es/index.mjs
import { c as r } from "./components.ts-chunk.mjs";
import a from "./_util/cssinjs/index.mjs";
import { _experimental as x } from "./_util/cssinjs/index.mjs";
import { default as n } from "./x-stream/x-stream.mjs";
import { default as c } from "./x-request/x-fetch.mjs";
import { default as h } from "./x-request/x-request.mjs";
import { XAgent as y, default as C } from "./use-x-agent/use-x-agent.mjs";
import { default as X } from "./use-x-chat/use-x-chat.mjs";
import { theme as T } from "ant-design-vue";
import { defaultPrefixCls as L, default as A } from "./x-provider/hooks/use-x-provider-context.mjs";
import { default as k } from "./_util/cssinjs/theme/Theme.mjs";
import { default as B } from "./_util/cssinjs/theme/createTheme.mjs";
import { extractStyle as q, default as E } from "./_util/cssinjs/hooks/useStyleRegister/index.mjs";
import { default as I } from "./_util/cssinjs/hooks/useCacheToken.mjs";
import { default as N } from "./_util/cssinjs/Keyframes.mjs";
import { default as W } from "./_util/cssinjs/transformers/legacyLogicalProperties.mjs";
import { default as w } from "./_util/cssinjs/transformers/px2rem.mjs";
import { default as D } from "./_util/cssinjs/linters/logicalPropertiesLinter.mjs";
import { default as H } from "./_util/cssinjs/linters/legacyNotSelectorLinter.mjs";
import { default as M } from "./_util/cssinjs/linters/parentSelectorLinter.mjs";
import { default as U } from "./_util/cssinjs/hooks/useCSSVarRegister.mjs";
import { X as Z } from "./x-provider/index-chunk.mjs";
import { default as ee } from "./bubble/index.mjs";
import { default as re } from "./bubble/BubbleList.mjs";
import { default as ae } from "./conversations/Conversations.mjs";
import { default as se } from "./welcome/Welcome.mjs";
import { default as me } from "./prompts/Prompts.mjs";
import { default as pe } from "./suggestion/Suggestion.mjs";
import { default as de } from "./sender/index.mjs";
import { default as ie } from "./attachments/index.mjs";
import { default as Se } from "./thought-chain/ThoughtChain.mjs";
import { default as ge } from "./actions/Actions.mjs";
import { StyleProvider as Ce, createCache as Pe, useStyleInject as Xe, useStyleProvider as ve } from "./_util/cssinjs/StyleContext.mjs";
import { token2CSSVar as be, unit as Le } from "./_util/cssinjs/util.mjs";
const f = function(e) {
  return Object.keys(r).forEach((o) => {
    const t = r[o];
    t.install && e.use(t);
  }), e.use(a.StyleProvider), e;
}, m = {
  install: f
};
export {
  ge as Actions,
  ie as Attachments,
  ee as Bubble,
  re as BubbleList,
  ae as Conversations,
  N as Keyframes,
  me as Prompts,
  de as Sender,
  Ce as StyleProvider,
  pe as Suggestion,
  k as Theme,
  Se as ThoughtChain,
  se as Welcome,
  y as XAgent,
  c as XFetch,
  Z as XProvider,
  h as XRequest,
  n as XStream,
  x as _experimental,
  Pe as createCache,
  B as createTheme,
  m as default,
  L as defaultPrefixCls,
  q as extractStyle,
  f as install,
  W as legacyLogicalPropertiesTransformer,
  H as legacyNotSelectorLinter,
  D as logicalPropertiesLinter,
  M as parentSelectorLinter,
  w as px2remTransformer,
  T as theme,
  be as token2CSSVar,
  Le as unit,
  U as useCSSVarRegister,
  I as useCacheToken,
  Xe as useStyleInject,
  ve as useStyleProvider,
  E as useStyleRegister,
  C as useXAgent,
  X as useXChat,
  A as useXProviderContext
};
